src/Umbraco.Infrastructure/Persistence/Repositories/Implement/ContentRepositoryBase.cs

```diff
@@ -1130,12 +1136,24 @@ protected void PersistRelations(TEntity entity)
                 }
                 else
                 {
-                    relations.Add(new ReadOnlyRelation(entity.Id, id, relationTypeId));
+                    relations.Add((id, relationTypeId));
                 }
             }
 
-            // Save bulk relations
-            RelationRepository.SaveBulk(relations);
+            // Get all existing relations (optimize for adding new and keeping existing relations)
+            var query = Query<IRelation>().Where(x => x.ParentId == entity.Id).WhereIn(x => x.RelationTypeId, relationTypeLookup.Values);
+            var existingRelations = RelationRepository.GetPagedRelationsByQuery(query, 0, int.MaxValue, out _, null)
+                .ToDictionary(x => (x.ChildId, x.RelationTypeId)); // Relations are unique by parent ID, child ID and relation type ID
+
+            // Add relations that don't exist yet
+            var relationsToAdd = relations.Except(existingRelations.Keys).Select(x => new ReadOnlyRelation(entity.Id, x.ChildId, x.RelationTypeId));
+            RelationRepository.SaveBulk(relationsToAdd);
+
+            // Delete relations that don't exist anymore
+            foreach (IRelation relation in existingRelations.Where(x => !relations.Contains(x.Key)).Select(x => x.Value))
+            {
+                RelationRepository.Delete(relation);
+            }
```

Comment:
This could be optimized to use a single `DELETE FROM ... WHERE ... IN (...)` query, but edits most likely won't be removing massive amounts of references and we already have an optimization at the top of the method when all references are removed. Also, looking at the `DeleteByParent()` implementation, this type of query requires a different solution for SQLite, which is sub-optimal (as it's a code smell to tightly couple the repository with specific database providers).